src/webhook/mutation/pod_mutator/dataingest_mutation/mutator.go

```diff
@@ -77,14 +76,14 @@ func (mutator *DataIngestPodMutator) ensureDataIngestSecret(request *dtwebhook.M
 		&endpointSecret)
 	if k8serrors.IsNotFound(err) {
 		err := endpointGenerator.GenerateForNamespace(request.Context, request.DynaKube.Name, request.Namespace.Name)
-		if err != nil {
-			log.Error(err, "failed to create the data-ingest endpoint secret before pod injection")
-			return errors.WithStack(err)
```

Comment:
since we log the error anyway, we could also wrap it with the info log messages.
could be harder to read though, but has the advantage of it being all in a single place

Reply:
the log lines shouldn't be far from each other, and logging the exact error twice is just spam.

I though about using `errors.WithMessage` but the problem with that is that there can only be 1 message for an error, so if multiple error handling `if`s add a message only the last (most hight level) one can be seen